### PR TITLE
Refactor commit. 

### LINKED ISSUE #31: API Architecture
We need to set up an architecture where we can easily facilitate Geomapping API validation and implementation of the rest of the sequence diagram. 

Some things to note/put in placeholders would be: other api calls where data like congressional districts, census tracts, etc. can be filled and verified (with decent or good error handling).

### PR DESCRIPTION
**Pull request by chrisong88**
Since it is a refactor commit, there are no changes to functionality. 

TL;DR: pulled out the surgeo and zrp into helper functions so that we can use them in other places in the future.

Refactored out the surgeo and zrp functionality into helper functions which can now be re-used together in the main API. Maintained the API endpoints to directly get surgeo and zrp predictions. These are useful for testing but may be deprecated down the line. Note: code is kind of unclean right now and needs more refactoring. I attempted a bigger refactor, but that has some bugs, so I wanted to get this one out for now.

I ran the old tests for the surgeo + zrp endpoints and they seem to work the same.

### CODE REVIEW

**Review by rnori-harv — APPROVED** [review]
I pulled and was able to run everything on the ZRP end smoothly (although it was calling the soon to be deprecated code). Nevertheless, the zrp() method that will be used in the main API seems to work smoothly and has the same logic / functionality as the code I uploaded. No huge changes here, so if Kayla's end still works correctly, feel free to merge.

**Review by rnori-harv — APPROVED** [review]
Accidentally left the comment for this PR on the next PR, but the same feedback is there. Definitely agree that moving the helper functions into the utils/ folder is a good idea as it makes for better modularization.

**Review by dzheng21 — APPROVED** [review]
